### ISSUE TITLE
OL8.2.0 GeoJSON readFeatures returns FeatureLike

### ISSUE THREAD

**Issue by Diaaz** [openlayers/openlayers#15334]
In OpenLayers 7 GeoJSON readFeatures returns Feature[] and these can be added to a VectorSource.

In OpenLayers 8.2 a FeatureLike[] is returned and that can be both Feature and RenderFeature. 

We have to work around that in all places that need Feature (and not RenderFeature) in our project with:
const features: Feature[] = new GeoJSON().readFeatures(json).filter(feature => feature instanceof Feature) as Feature[];

Is this intended? Would it be simpler to use a specific RenderFeature reader?

**Comment by ahocevar** [comment]
The GeoJSON format returns ol/Feature instances, unless you configure it with `featureClass: RenderFeature`. So no need to do an `instanceof` check. Or are you a TypeScript user? Generics could be used to make your life easier here. Do you want to create a pull request?

**Comment by ahocevar** [comment]
For the GeoJSON format we can solve this with generics. For VectorTile's `getFeaturesInExtent()` we cannot, because we cannot know whether you feed `Feature` or `RenderFeature` instances to the `VectorTile` source. And converting between the two is expensive, which is the reason why we introduced `RenderFeature` in the first place.

I'd be interested to hear which parts of the `Feature` API you're using that's not covered by `FeatureLike`. Usually, if you use the `FeatureLike` type in your application, you should be fine.

**Comment by ahocevar** [comment]
#15338 improves `GeoJSON#readFeatures()` to report the correct return type, based on the `featureClass` constructor property.

**Comment by Siedlerchr** [comment]
So far this is only solved for readFeatures. 
Reading a single Feature still returns FeatureLike or RenderFeature[]  
My use case is the following: 

``` 
    const myFeature = new GeoJSON().readFeature(e.search.geojson)
    new VectorSource().addFeature(myFeature)

```
Argument of type 'FeatureLike | RenderFeature[]' is not assignable to parameter of type 'Feature<Geometry>'.
  Type 'RenderFeature' is missing the following properties from type 'Feature<Geometry>': on, once, un, geometryName_, and 39 more.

**Comment by ahocevar** [comment]
@Siedlerchr True. Would you be able to create a pull request? I think it should be straightforward using the generics that were added with this pull request.

**Comment by ahocevar** [comment]
@Siedlerchr I went ahead and fixed this with #15377.

**Comment by Siedlerchr** [comment]
Thanks so much!

**Comment by alecarn** [comment]
@ahocevar is this released in v8.2.0?

**Comment by alecarn** [comment]
Thanks we are waiting for this one to be deployed to upgrade to v8. This was a blocker for our Typescript project.
